media/js/mozorg/home/takeover-2016.js

```diff
@@ -0,0 +1,124 @@
+    var $donateCustom = $('#donate-custom');
+    var $optionLabels = $('label[class!="own-amount"]', $donateOptionsSection);
+    var $options = $('input[type="radio"]', $donateOptionsSection);
+
+    // sets the mozorg.takeover state to true indicating that the user has
+    // seen the takeover modal during their current session.
+    function storeSession() {
+        try {
+            sessionStorage.setItem('mozorg.takeover.seen', 'true');
+        } catch (ex) {
+            // yum, errors taste nice ;)
+        }
+    }
+
+    // Show and hides the takeover modal
+    // Removes the keyup event from body if specified
+    function toggleModal(removeListener) {
+        if(removeListener) {
+            // no longer listen for keyup events.
+            $body.off('keyup.takeover');
+        }
+
+        $([$takeoverModal, $overlay]).each(function() {
+            $(this).toggleClass('hidden');
+        });
+    }
+
+    // removes checked from all the radio buttons
+    // and selected from the labels.
+    function clearOptions() {
+        $optionLabels.removeClass('selected');
+        $options.removeAttr('checked');
+    }
+
+    // shows the take over modal, sets the required entry in
+    // session storage and adds an `esc` key handler to close
+    // the modal.
+    function initTakeOver() {
+        // show the takeover
+        toggleModal();
+        // set mozorg.takeover.seen to true
+        storeSession();
+
+        // trap key events, listening for escape.
+        // we only want to do this here so, we do not attach the event
+        // if we are not showing the modal.
+        $body.on('keyup.takeover', function(event) {
+            if (event.key === 'Escape' || event.keyCode === 27) {
+                // close the modal
+                toggleModal(true);
+            }
+        });
+
+        $overlay.on('click', function(){
+            toggleModal();
+        });
+    }
+
+    // handles clicks on the labels of the radio input elements.
+    $optionLabels.on('click', function() {
+        // clear selected class from labels
+        $optionLabels.removeClass('selected');
+        // empty any value that might have been typed
+        // into the custom amount field.
+        $donateCustom.val('');
+        // set the clicked label to selected
+        $(this).addClass('selected');
+    });
+
+    // once a user types into the custom amount field
+    // clear all other options.
+    $donateCustom.keyup(clearOptions);
+
+    // the name of the parameter that indicates the amount the user
+    // wish to donate has to be called amount. Seeing that we have
+    // radio buttons with that name as well as an input field, we need
+    // to disable the input field prior to form submission to avoid a
+    // scenario where we have ?amount=50&amount=
+    $form.on('submit', function() {
+        if ($donateCustom.val() === '') {
+            $donateCustom.attr('disabled', 'disabled');
```

Comment:
That's a tricky one. We need to disable it on submit to prevent sending the blank amount, but upon using the back button there isn't any sort of event to trigger a reset. My first thought is a timer that just occasionally checks if the field is disabled and re-enables it if so, but that seems like a bad idea... any thoughts on a simple solution or should we just let it slide?